src/hotspot/share/utilities/exceptions.cpp

```diff
@@ -205,7 +205,7 @@ void Exceptions::_throw_msg_cause(JavaThread* thread, const char* file, int line
 void Exceptions::_throw_cause(JavaThread* thread, const char* file, int line, Symbol* name, Handle h_cause,
                               Handle h_loader, Handle h_protection_domain) {
   // Check for special boot-strapping/compiler-thread handling
-  if (special_exception(thread, file, line, h_cause)) return;
```

Comment:
This fixes a long standing bug `special_exception` is being queried with the *cause* of the exception being thrown instead of the *name* of the exception being thrown.

Reply:
I'm not so sure this is in fact a bug. If we are throwing with a cause, but we can't actually throw and so will do vm_exit, then the exception of interest is the cause not the more generic exception that would otherwise contain the cause.

Though I have to wonder why there is not an original `_throw` for the "cause" exception, that would have triggered the special_exception handling anyway?

Reply:
Though I see this is inconsistent with `Exceptions::_throw_msg_cause`

Reply:
Great. Would you mind approving this PR as this is the only non-JVMCI file changed.